packages/actor-init-sparql/lib/ActorInitSparql-browser.ts

```diff
@@ -31,6 +36,49 @@ export class ActorInitSparql extends ActorInit implements IActorInitSparqlArgs {
     super(args);
   }
 
+  /**
+   * Create a copy of the given operation in which all given bindings are applied.
+   * The bindings are applied to all quad patterns and path expressions.
+   *
+   * @param {Operation} operation An operation.
+   * @param {Bindings} initialBindings Bindings to apply.
+   * @return {Operation} A copy of the given operation where all given bindings are applied.
+   */
+  public static applyInitialBindings(operation: Algebra.Operation, initialBindings: Bindings): Algebra.Operation {
+    const copiedOperation: Algebra.Operation = <any> {};
+    for (const key of Object.keys(operation)) {
+      if (Array.isArray(operation[key])) {
+        if (key === 'variables') {
+          copiedOperation[key] = operation[key].filter(
+            (variable: RDF.Variable) => initialBindings.keySeq().indexOf(termToString(variable)) < 0);
+        } else {
+          copiedOperation[key] = operation[key].map(
+            (subOperation: Algebra.Operation) => ActorInitSparql.applyInitialBindings(subOperation, initialBindings));
```

Comment:
It says `subOperation: Algebra.Operation` but this is not always the case though. There are also operations that have an array of terms as value. Not sure if any of those operations can actually appear at this point though (operations as Describe, From, Nps (part of path expressions) and Values)